src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        # Tests most basic post call, finish_mega_field (can be called multiple times without causing a problem)
+        expected_status_code = 204
+        status_code = self.ASM_manager.asmApiPostCall("/scan/finish_mega_field", expected_status_code)
+        self.assertEqual(status_code, expected_status_code)
+
+    def test_externalStorageURL_VA(self):
+        # Setting URL
+        test_url = 'ftp://testname:testword@testable.com/Test_images'
+        self.ASM_manager.externalStorageURL.value = test_url
+        self.assertEqual(self.ASM_manager.externalStorageURL.value, test_url)
+
+        # Test Scheme
+        with self.assertRaises(ValueError):
+            self.ASM_manager.externalStorageURL.value = 'wrong://testname:testword@testable.com/Test_images'
+        self.assertEqual(self.ASM_manager.externalStorageURL.value, test_url)
+
+        # Test User
+        with self.assertRaises(ValueError):
+            self.ASM_manager.externalStorageURL.value = 'ftp://wrong%user:testword@testable.com/Test_images'
+        self.assertEqual(self.ASM_manager.externalStorageURL.value, test_url)
+
+        # Test Password
+        with self.assertRaises(ValueError):
+            self.ASM_manager.externalStorageURL.value = 'ftp://testname:testwrong%$word@testable.com/Test_images'
+        self.assertEqual(self.ASM_manager.externalStorageURL.value, test_url)
+
+        # Test Host
+        with self.assertRaises(ValueError):
+            self.ASM_manager.externalStorageURL.value = 'ftp://testname:testword@non-test-%-able.com/Test_images'
+        self.assertEqual(self.ASM_manager.externalStorageURL.value, test_url)
+
+        # Test Path
+        with self.assertRaises(ValueError):
+            self.ASM_manager.externalStorageURL.value = 'ftp://testname:testable.com/Inval!d~Path'
+        self.assertEqual(self.ASM_manager.externalStorageURL.value, test_url)
+
+    def test_assembleCalibrationMetadata(self):
+        MAX_NMBR_POINT = 9000  # Constant maximum number of setpoints
+        # TODO MAX_NMBR_POINT value of 9000 is not working for the entire range of the dwell time, this value
+        #  needs to be increased so that the entire range can be tested. Then the multipication of 0.99 via FIX in line
+        #  178  can be deleted
+        FIX = 0.99
+
+        descanner = self.MirrorDescanner
+        scanner = self.EBeamScanner
+        ASM = self.ASM_manager
+        mppc = self.MPPC
+
+        # Test over the entire range many options to also let possible floating point error surface
+        for test_repetition in REPEAT_1000:
+            # Check randomly all the options of the dwell time allowed.
+            minimum_dwell_time = scanner.dwellTime.range[0]
+            random_dwell_time = numpy.round(numpy.random.random() * FIX * scanner.dwellTime.range[1], 9)
+            scanner.dwellTime.value = max(random_dwell_time, minimum_dwell_time)
+
+            line_scan_time = scanner.dwellTime.value * mppc.cellCompleteResolution.value[0]
```

Comment:
Otherwise you cannot check the results. Since the implementation of _assembleCalibrationMetadata() might chance (and if fact already did). This test will check will see if a new implementation is still correct or if something changed.

Reply:
ok